rtloader/two/two.cpp

```diff
@@ -529,6 +529,137 @@ char **Two::getCheckWarnings(RtLoaderPyObject *check)
     return warnings;
 }
 
+diagnoses_t *Two::getCheckDiagnoses(RtLoaderPyObject *check)
```

Comment:
This looks like the same implementation between `two` and `three`. We might want to move logic to a helper function ?

Reply:
I can do it and I was not particular thrilled about it but it loos like bunch of two and three methods are mostly copy/paste. I do not know if it is a pattern I should observe and if there are good reasons for that and I was trying to follow existing pattern. Correct me if I am wrong here.

Reply:
The issue with other method was that there were just enough difference between two and three. In the end making shared code between the two was more trouble that it was worth it. But in this case the code seems exactly the same (I guess Python list API didn't changed from 2 to 3). I think it's worth making an helper function.

Reply:
Ok, I will move these virtual methods in the base class.

Reply:
Looks it is a bit involved for comfort in terms of include dependencies. Will move common code into common/diagnose.c/h files

Reply:
That was painful. I have tried a few things. I could not move to the base because it appears to be Python version neutral and I could not have a general rtloader general CPP code. I have ended up creating two lower level size calculation and serialization functions. I did not eliminate duplicate code completely by adding a good chunk of it to the util code. Let me know if it is good that I can resolve the comment.